src/sections/mod.rs

```diff
@@ -119,12 +119,17 @@ impl<'a> PsdCursor<'a> {
     }
 
     /// Get the cursor's position
-    pub fn position(&self) -> u64 {
-        self.cursor.position()
+    pub fn position(&self) -> usize {
+        // since we are a cursor over a [u8]
+        // that means the total amount of bytes is less than isize::MAX
+        // therefore the biggest index is isize::MAX
+        // also meaning that it can be safely cast to a usize
```

Comment:
Let's clarify why this is true.

As in, why does a cursor over a `[u8]` have `isize::MAX` bytes?

Reply:
from [the standard library](https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html) it states:

*The total `size len * mem::size_of::<T>()` of the slice must be no larger than `isize::MAX`, and adding that size to data must not “wrap around” the address space. See the safety documentation of [pointer::offset](https://doc.rust-lang.org/std/primitive.pointer.html#method.offset).

and this also applies to all types, all allocations/types/abstract machine objects have to not wrap around the address space, so anything in memory must not be larger than `isize::MAX` bytes